src/EffectBase.cpp

```diff
@@ -116,14 +123,44 @@ void EffectBase::SetJson(const std::string value) {
 }
 
 // Load Json::Value into this object
-void EffectBase::SetJsonValue(const Json::Value root) {
+void EffectBase::SetJsonValue(Json::Value root) {
+
+	if (ParentTimeline()){
+		// Get parent timeline
+		Timeline* parentTimeline = (Timeline *) ParentTimeline();
+
+		// Get the list of effects on the timeline
+		std::list<EffectBase*> effects = parentTimeline->ClipEffects();
+
+		// Loop through the effects
+		for (auto const& effect : effects){
+			// Set the properties of all effects which parentEffect points to this
+			if (effect->info.parent_effect_id == this->Id())
+				effect->SetJsonValue(root);	
+		}
+	}
+
+	// Set this effect properties with the parent effect properties (except the id and parent_effect_id)
+	if (parentEffect){
+		root = parentEffect->JsonValue();
```

Comment:
This feels very weird to me. So if we imagine I have an `EffectBase* blur` that points to my blur effect, and I call its `SetJson()` like this:
```cpp
blur->SetJson(R"json(
    "iterations": {
      "Points": [
        { "co": { "X": 1.0, "Y": 2.0 }, "interpolation": 1 }
       ] }
   )json");
```
Then whatever data I **actually pass in** to `SetJson()` or `SetJsonValue()` will be _ignored_ if there's a parent effect set, in favor of just re-duplicating whatever the parent effect's settings are?

If `SetJsonValue()` is going to ignore its own arguments when it's called on a parented effect, should it maybe just raise an exception instead? There doesn't seem to be any clear value to calling it. (Or maybe it should have a no-arguments overload that triggers a refresh of the parent effect data, if that's what calling it is intended to achieve.)